src/snowflake/connector/cursor.py

```diff
@@ -1724,13 +1724,38 @@ def get_result_batches(self) -> list[ResultBatch] | None:
 
 class DictCursor(SnowflakeCursor):
     """Cursor returning results in a dictionary."""
-
     def __init__(self, connection) -> None:
         super().__init__(
             connection,
             use_dict_result=True,
         )
 
+    def fetchone(self) -> dict | None:
+        return super().fetchone()
+    
+    def fetchmany(self, size: int | None = None) -> list[dict]:
+        return super().fetchmany()
+    
+    def fetchall(self) -> list[dict]:
+        return super().fetchall()
+
+class TupleCursor(SnowflakeCursor):
```

Comment:
I don't think we need a TupleCursor here -- this is the effectively the default SnowflakeCursor